datafusion/spark/src/function/datetime/make_interval.rs

```diff
@@ -43,8 +43,39 @@ impl Default for SparkMakeInterval {
 
 impl SparkMakeInterval {
     pub fn new() -> Self {
+        fn int32_coercion() -> Coercion {
+            Coercion::new_implicit(
+                TypeSignatureClass::Native(logical_int32()),
+                vec![TypeSignatureClass::Integer],
+                NativeType::Int32,
+            )
+        }
```

Comment:
```suggestion
        let int32 = Coercion::new_implicit(
            TypeSignatureClass::Native(logical_int32()),
            vec![TypeSignatureClass::Integer],
            NativeType::Int32,
        )
```

Probably can just have them as variables and clone where necessary